client/lib/media/test/utils.js

```diff
@@ -84,6 +84,40 @@ describe( 'MediaUtils', function() {
 		} );
 	} );
 
+	describe( '#getFileExtension()', function() {
+		it( 'should return undefined for a falsey media value', function() {
+			expect( MediaUtils.getFileExtension() ).to.be.undefined;
+		} );
+
+		it( 'should detect extension from file name', function() {
+			expect( MediaUtils.getFileExtension( 'example.gif' ) ).to.equal( 'gif' );
+		} );
+
+		it( 'should ignore querystring parameters', function() {
+			expect( MediaUtils.getFileExtension( 'example.gif?w=100' ) ).to.equal( 'gif' );
+		} );
+
+		it( 'should detect extension from HTML5 File object', function() {
+			expect( MediaUtils.getFileExtension( new window.File( [''], 'example.gif' ) ) ).to.equal( 'gif' );
```

Comment:
So, I was a bit confused why Babel didn't throw up at this line*, and it seems to be the consequence of some spillover from the `actions.js` test where a fake DOM is initialized but never cleaned up. Once we migrate these tests to a single test runner, this will become apparent in its breakage.

You can see the issue when running the test in isolation with Mocha from the `client/lib/media` folder:

```bash
media (fix/3817-upload-media-url-querystring) $ NODE_ENV=test NODE_PATH=../../../client mocha --compilers js:babel/register test/utils.js 


  MediaUtils
    #getFileExtension()
      1) should detect extension from HTML5 File object


  0 passing (559ms)
  1 failing

  1) MediaUtils #getFileExtension() should detect extension from HTML5 File object:
     ReferenceError: window is not defined
      at Context.<anonymous> (/Users/andrew/Documents/Code/wp-calypso/client/lib/media/test/utils.js:101:21)
      at callFn (/usr/local/lib/node_modules/mocha/lib/runnable.js:250:21)
      at Test.Runnable.run (/usr/local/lib/node_modules/mocha/lib/runnable.js:243:7)
      at Runner.runTest (/usr/local/lib/node_modules/mocha/lib/runner.js:373:10)
      at /usr/local/lib/node_modules/mocha/lib/runner.js:451:12
      at next (/usr/local/lib/node_modules/mocha/lib/runner.js:298:14)
      at /usr/local/lib/node_modules/mocha/lib/runner.js:308:7
      at next (/usr/local/lib/node_modules/mocha/lib/runner.js:246:23)
      at Immediate._onImmediate (/usr/local/lib/node_modules/mocha/lib/runner.js:275:5)
      at processImmediate [as _immediateCallback] (timers.js:383:17)
```

The resolution should be fairly straight-forward. Despite this test not being part of the single test runner, you should still be able to use the [`use-fake-dom` helper](https://github.com/Automattic/wp-calypso/blob/master/test/test/helpers/use-fake-dom/README.md) to initialize a DOM for the duration of this test.

_*_ Because, by default, DOM objects aren't available unless a fake DOM is created.